src/app/components/cohort-view/cohort-view.directive.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/**
+ * @memberof transmartBaseUi
+ * @ngdoc directive
+ * @name cohortView
+ */
+angular.module('transmartBaseUi')
+    .directive('cohortView', function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'app/components/cohort-view/cohort-view.tpl.html',
+            controller: ['$scope', '$timeout', '$q', 'CohortViewService', 'AlertService',
+                function ($scope, $timeout, $q, CohortViewService, AlertService) {
+
+                    $scope.cohorts = [];
+                    var ctrl = this;
+
+                    ctrl.gridOptions = CohortViewService.options;
+
+                    /** Loads the list of cohorts.
+                     */
+                    ctrl.loadCohorts = function() {
+                        CohortViewService.getCohorts().then(function (cohorts) {
+                            $scope.cohorts = cohorts;
+                        });
```

Comment:
In    CohortViewService.getCohorts you defined deferred.reject. What is your plan to it I didnt see you use it here?